### ISSUE TITLE
Expected catch() or return for Promises?

### ISSUE THREAD

**Issue by pdehaan** [mozilla/activity-stream#802]
I randomly found [**eslint-plugin-promise**](https://www.npmjs.com/package/eslint-plugin-promise) and it throws a couple warnings/errors where our Promise chains don't have a `.catch()` statement, in case a promise fails.

Not sure if we care, so I'll file this bug and y'all can discuss.
### Steps to reproduce:
1. `$ npm i eslint-plugin-promise -D`
2. Edit .eslintrc and add `"promise"` to `plugins:[]` array:
   
   ``` js
   "plugins": [
     "mozilla",
     "promise",
     "react"
   ],
   ```
3. Add following rules to .eslintrc:
   
   ``` js
   promise/param-names: 2
   promise/always-return: 2
   promise/catch-or-return: 2
   promise/no-native: 0
   ```
4. `$ npm run test:lint`
### Actual results:

``` bash
$ npm run test:lint

> activity-streams@1.0.19 test:lint /Users/pdehaan/dev/github/mozilla/activity-streams
> eslint . && sass-lint -v -q


/Users/pdehaan/dev/github/mozilla/activity-streams/lib/ActivityStreams.js
  168:9  error  Expected catch() or return                  promise/catch-or-return
  168:9  error  Each then() should return a value or throw  promise/always-return
  173:9  error  Expected catch() or return                  promise/catch-or-return
  173:9  error  Each then() should return a value or throw  promise/always-return
  178:9  error  Expected catch() or return                  promise/catch-or-return
  178:9  error  Each then() should return a value or throw  promise/always-return
  183:9  error  Expected catch() or return                  promise/catch-or-return
  183:9  error  Each then() should return a value or throw  promise/always-return
  223:5  error  Expected catch() or return                  promise/catch-or-return
  235:9  error  Expected catch() or return                  promise/catch-or-return
  235:9  error  Each then() should return a value or throw  promise/always-return

/Users/pdehaan/dev/github/mozilla/activity-streams/lib/main.js
  70:7  error  Each then() should return a value or throw  promise/always-return

/Users/pdehaan/dev/github/mozilla/activity-streams/lib/PlacesProvider.js
  229:9  error  Expected catch() or return                  promise/catch-or-return
  229:9  error  Each then() should return a value or throw  promise/always-return
  243:9  error  Expected catch() or return                  promise/catch-or-return
  243:9  error  Each then() should return a value or throw  promise/always-return

/Users/pdehaan/dev/github/mozilla/activity-streams/lib/PreviewProvider.js
  154:11  error  Expected catch() or return                  promise/catch-or-return
  154:11  error  Each then() should return a value or throw  promise/always-return

/Users/pdehaan/dev/github/mozilla/activity-streams/lib/TabTracker.js
  279:5  error  Expected catch() or return                  promise/catch-or-return
  279:5  error  Each then() should return a value or throw  promise/always-return
  280:5  error  Expected catch() or return                  promise/catch-or-return
  280:5  error  Each then() should return a value or throw  promise/always-return

/Users/pdehaan/dev/github/mozilla/activity-streams/lib/TelemetrySender.js
  66:5  error  Each then() should return a value or throw  promise/always-return

/Users/pdehaan/dev/github/mozilla/activity-streams/test/test-ActivityStreams-workers.js
  90:3  error  Expected catch() or return  promise/catch-or-return

✖ 24 problems (24 errors, 0 warnings)
```

Could be interesting to add some tracking in `.catch()` to see how often (if ever) the Promise fails for whatever reason.

**Comment by pdehaan** [comment]
Not sure if this would be handled via #747; _"Add Sentry tooling to front-end"_, or if we'd still need to explicitly catch a Promise error and pipe it through Sentry.

**Comment by sarracini** [comment]
This was fixed with #1667
